### PR TITLE
Fix inconsistent pressure units in HEMCO standalone

### LINKED ISSUE #277: Incorrect handling of pressure unit in HEMCO standalone
### Your name

Yuanjian Zhang

### Your affiliation

WashU

## Issue
HEMCO standalone, different from HEMCO within geos-chem, has a internal pressure unit of Pa instead of hPa 
(See [geos-chem/GeosCore/hco_interface_gc_mod.F90](https://github.com/geoschem/geos-chem/blob/1ef3d7f88da95ca2425267fc4f96bd1524fc985f/GeosCore/hco_interface_gc_mod.F90#L3326))

Processed meteorological fields for MERRA-2 and GEOS-FP are separately in Pa and hPa.

Unlike geos-chem, HEMCO standalone does not have the automatic met-specific unit conversion in 
[geos-chem/GeosCore/flexgrid_read_mod.F90](https://github.com/geoschem/geos-chem/blob/1ef3d7f88da95ca2425267fc4f96bd1524fc985f/GeosCore/flexgrid_read_mod.F90#L1284-L1289)

So pressure is mistakenly treated as Pa if using GEOS-FP(and GEOS-IT) to drive HEMCO standalone. Possibly making output emission field in kg/m2/s 1000 times smaller.

Related issues: #275

## Workaround
Manually use a `hPa2Pa` scale factor of 100.0 in configuration.

But I also wish this can be addressed in a more natural manner like in GC-Classic and GCHP.

### What are the steps to reproduce the bug?

use standard run directory creation script to run a HEMCO standalone instance with GEOS-FP

### Please attach any relevant configuration and log files.

_No response_

### What HEMCO version were you using?

standalone

### What environment were you running HEMCO on?

Local cluster

### What compiler and version were you using?

gcc 10.2.0

### Will you be addressing this bug yourself?

Yes, but I will need some help

### In what configuration were you running HEMCO?

Standalone

### As what resolution were you running HEMCO?

all

### What meterology fields did you use?

Other (please explain in additional information section below)

### Additional information

GEOS-FP and GEOS-IT

### PR DESCRIPTION
**Pull request by yantosca**
### Name and Institution (Required)

Name: Bob Yantosca
Institution: Harvard + GCST

### Describe the update
This is the companion PR to #277 by @yuanjianz.  It fixes the inconsistent pressure unit handling in the HEMCO standalone as described in that issue.

### Expected changes
The HEMCO standalone will use 'Pa' for pressure units regardless of input meteorology.

### Related Github Issue
- Closes #277

### CODE REVIEW

File: src/Core/hco_geotools_mod.F90 [hunk 1]
```diff
@@ -1178,11 +1179,26 @@ SUBROUTINE HCO_CalcVertGrid ( HcoState, PSFC, ZSFC, TK, BXHEIGHT, PEDGE, RC )
...
+    ! If we are using HEMCO in a CTM or ESM, then PEDGE will have
+    ! been passed to HEMCO as an input.  If PEDGE has not been passed
+    ! (e.g. HEMCO standalone), then compute it here using the surface
+    ! pressure PSFC and the Ap and Bp hybrid grid parameters.
     IF ( .NOT. FoundPEDGE ) THEN
        !$OMP PARALLEL DO        &
        !$OMP DEFAULT( SHARED  ) &
-       !$OMP PRIVATE( I, J, L )
+       !$OMP PRIVATE( I, J, L ) &
+       !$OMP COLLAPSE( 3      )
```
Comment (yantosca) [inline src/Core/hco_geotools_mod.F90:1201]: Collapsing the loop will result in a speedup.

**Comment by yantosca** [comment]
Integration tests are running.

**Comment by yantosca** [comment]
All GEOS-Chem Classic integration tests passed:
```console
==============================================================================
GEOS-Chem Classic: Execution Test Results

GCClassic #6f22240 GEOS-Chem submod update: Merge PR #2398 (Uncomment cations for HETP)
GEOS-Chem #e29a4c928 Merge PR #2398 (Pass non-zero Ca2, Mg, and K cations to HETP)
HEMCO     #8be6b60 Convert PSFC from hPa -> Pa in HCO_CalcVertGrid

Using 24 OpenMP threads
Number of execution tests: 28

Submitted as SLURM job: 42183358
==============================================================================

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%  All execution tests passed!  %%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```
Also, all tests were zero-diff w/r/t the prior PR #2398 except for 
- TOMAS (parallelization error?)
- APM (parallelization error?)

**Comment by yantosca** [comment]
All GCHP integration tests passed:
```console
==============================================================================
GCHP: Execution Test Results

GCHP      #136fa70 GEOS-Chem submod update: Merge PR #2398 (Uncomment cations for HETP)
GEOS-Chem #e29a4c928 Merge PR #2398 (Pass non-zero Ca2, Mg, and K cations to HETP)
HEMCO     #

Number of execution tests: 11

Submitted as SLURM job: 42183723
==============================================================================
 
Execution tests:
------------------------------------------------------------------------------
gchp_merra2_carbon..................................Execute Simulation....PASS
gchp_merra2_carbon_CH4..............................Execute Simulation....PASS
gchp_merra2_carbon_CO...............................Execute Simulation....PASS
gchp_merra2_carbon_CO2..............................Execute Simulation....PASS
gchp_merra2_carbon_OCS..............................Execute Simulation....PASS
gchp_merra2_fullchem................................Execute Simulation....PASS
gchp_merra2_fullchem_benchmark......................Execute Simulation....PASS
gchp_merra2_fullchem_RRTMG..........................Execute Simulation....PASS
gchp_merra2_fullchem_TOMAS15........................Execute Simulation....PASS
gchp_merra2_tagO3...................................Execute Simulation....PASS
gchp_merra2_TransportTracers........................Execute Simulation....PASS
 
Summary of test results:
------------------------------------------------------------------------------
Execution tests passed: 11
Execution tests failed: 0
Execution tests not yet completed: 0

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%  All execution tests passed!  %%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```
Also, all tests were zero-diff w/r/t PR #2398.

**Review by msulprizio — APPROVED** [review]
